dubbo-rpc/dubbo-rpc-api/src/main/java/org/apache/dubbo/rpc/model/ConsumerModel.java

```diff
@@ -29,11 +29,13 @@
 public class ConsumerModel {
     private final Object proxyObject;
     private final String serviceName;
+    private final Class<?> serviceInterfaceClass;
 
     private final Map<Method, ConsumerMethodModel> methodModels = new IdentityHashMap<Method, ConsumerMethodModel>();
 
-    public ConsumerModel(String serviceName, Object proxyObject, Method[] methods, Map<String, Object> attributes) {
+    public ConsumerModel(String serviceName, Class<?> serviceInterfaceClass, Object proxyObject, Method[] methods, Map<String, Object> attributes) {
         this.serviceName = serviceName;
+        this.serviceInterfaceClass = serviceInterfaceClass;
         this.proxyObject = proxyObject;
 
         if (proxyObject != null) {
```

Comment:
Is methods and attributes are optional and can be consider my this class only if proxyObject not null? If this is the case then can we have two constructor one with only service and another one with all the parameters?

Reply:
I am afraid I don't fully understand, could you pls. rephrase your concern?

Reply:
In fact, `if (proxyObject != null)` is a defensive programming. In Dubbo, proxyObject should not be null and will not be null. 

I guess we should change it to `assert` or `throw IllegelArgumentException`, would you mind take care of this in other change? 

This change contains too much already, in my opinion.

Reply:
Sure.